utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/azure/AzureSchemaRegistryClientFactory.scala

```diff
@@ -133,10 +135,8 @@ class AzureSchemaRegistryClient(config: SchemaRegistryClientKafkaConfig) extends
 
   private def getOneMatchingSchemaName(topicName: String, isKey: Boolean): Validated[SchemaRegistryError, String] = {
     getAllFullSchemaNames.andThen { fullSchemaNames =>
-      val matchingFullSchemaNames = fullSchemaNames.collect {
-        case fullSchemaName @ FullSchemaNameDecomposed(`topicName`, _, `isKey`) =>
-          fullSchemaName
-      }
+      val matchStrategy           = SchemaNameTopicMatchStrategy(List(topicName))
```

Comment:
It looks tricky for me, `matchAllSchemas` is always created with `SchemaNameTopicMatchStrategy` with fixed one element list of topic names. I think that it would be easier to have a single method in the `SchemaNameTopicMatchStrategy` object. let's say `getMatchingChemaNames(topicName, fullSchemaNames, isKey)`. WDYT?

Reply:
fixed